src/yachtzee/index.ts

```diff
@@ -1,163 +1,116 @@
-import { potionSetup } from "../potions";
-import { prepRobortender } from "../tasks/dailyFamiliars";
-import { yachtzeePotionSetup } from "./buffs";
-import { executeNextDietStep, yachtzeeChainDiet } from "./diet";
-import { pyecAvailable, shrugIrrelevantSongs } from "./lib";
-import {
-  getBestWaterBreathingEquipment,
-  maximizeMeat,
-  prepareOutfitAndFamiliar,
-  stickerSetup,
-} from "./outfit";
+import { bestYachtzeeFamiliar } from "./familiar";
+import { waterBreathingEquipment } from "../outfit";
 
-function _yachtzeeChain(): void {
-  if (!canInteract()) return;
-  // We definitely need to be able to eat sliders and drink pickle juice
-  if (!realmAvailable("sleaze")) return;
+function yachtzeeTasks(equipment: Item): Task[] {
+  const familiar = bestYachtzeeFamiliar();
 
-  maximize("MP", false);
-  meatMood(false, 750 + baseMeat).execute(embezzlerCount());
-  potionSetup(globalOptions.nobarf); // This is the default set up for embezzlers (which helps us estimate if chaining is better than extros)
-  maximizeMeat();
-  prepareOutfitAndFamiliar();
+  const meatOutfit = toSpec(
+    new Requirement(
+      [
+        "meat",
+        ...(familiar.underwater || have($effect`Driving Waterproofly`) || have($effect`Wet Willied`)
+          ? []
+          : ["underwater familiar"]),
+      ],
+      {
+        forceEquip: [equipment],
+        preventEquip: $items`anemoney clip, cursed magnifying glass, Kramco Sausage-o-Matic™, cheap sunglasses`,
+        modes: equipment === $item`The Crown of Ed the Undying` ? { edpiece: "fish" } : {},
+      },
+    ),
+  );
 
-  const meatLimit = 5000000;
-  if (myMeat() > meatLimit) {
-    const meatToCloset = myMeat() - meatLimit;
-    print("");
-    print("");
-    print(`We are going to closet all-but-5million meat for your safety!`, "blue");
-    print("");
-    print("");
-    if (!get("_yachtzeeChainClosetedMeat")) {
-      set("_yachtzeeChainClosetedMeat", meatToCloset);
-    } else {
-      set("_yachtzeeChainClosetedMeat", meatToCloset + get("_yachtzeeChainClosetedMeat"));
-    }
-    cliExecute(`closet put ${meatToCloset} meat`);
-  }
-  if (!yachtzeeChainDiet()) {
-    if (get("_yachtzeeChainClosetedMeat", 0)) {
-      cliExecute(`closet take ${get("_yachtzeeChainClosetedMeat")} meat`);
-    }
-    set("_yachtzeeChainClosetedMeat", 0);
-    return;
-  }
-  let jellyTurns = get("_stenchJellyChargeTarget", 0);
-  let fishyTurns = haveEffect($effect`Fishy`) + (pyecAvailable() ? 5 : 0);
-  let turncount = myTurncount();
-  yachtzeePotionSetup(Math.min(jellyTurns, fishyTurns));
-  stickerSetup(Math.min(jellyTurns, fishyTurns));
-  if (get("_yachtzeeChainClosetedMeat", 0)) {
-    cliExecute(`closet take ${get("_yachtzeeChainClosetedMeat")} meat`);
-  }
-  set("_yachtzeeChainClosetedMeat", 0);
-  if (haveEffect($effect`Beaten Up`)) {
-    uneffect($effect`Beaten Up`);
-  }
-  meatMood(false, 2000).execute(Math.min(jellyTurns, fishyTurns));
-  safeRestore();
+  return [
```

Comment:
This needs handling for digitized embezzlers, which will ruin our parade. We dont' want to free run from them and we don't want to hit them in the sunken party yacht.

The smartest and coolest thing to do would be to add a check to the embezzler underwater code that says that if we have SBB and NC forcers, we _don't_ get fishy during that stint, and we instead assume that we'll do the envyfish later in the day.

It would be nice if we started saving an envyfish egg for the next day. Is that a thing we can do?

Reply:
Also kramco interpolation

Reply:
Also romanced embezzlers :-/